### PR TITLE
Upgrade GKE package to kpt v1

### PR DESCRIPTION
**Pull request by karlkfi**

### CODE REVIEW

File: catalog/gke/cluster/README.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  - [ContainerCluster](https://cloud.google.com/config-connector/docs/reference/resource-docs/container/containercluster) to configure a GKE cluster.
+- [container-api.yaml](/catalog/gke/cluster/container-api.yaml)
+  - [Service](https://cloud.google.com/config-connector/docs/reference/resource-docs/serviceusage/service) to enable the container.googleapis.com API.
+
+## Setters
+
+Setters are inherited by sub-packages.
+
+```
+$ kpt cfg list-setters .
```
Comment (bharathkkb) [inline catalog/gke/cluster/README.md:17]: I don't think we have a replacement in v1 yet
https://github.com/GoogleContainerTools/kpt/issues/2097

File: catalog/gke/cluster/setters.yaml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  # The project that the VPC is in
+  network-project-id: network-project-id # kpt-set: ${network-project-id}
+  # The namespace in which to reference the subnet resource
+  networking-namespace: networking # kpt-set: ${networking-namespace}
+  # The namespace in which to manage cluster resources
+  platform-namespace: config-control # kpt-set: ${platform-namespace}
+  # The project in which to manage cluster resources
+  platform-project-id: platform-project-id # kpt-set: ${platform-project-id}
+  # The namespace in which to manage service enablement resources
+  projects-namespace: projects # kpt-set: ${projects-namespace}
```
Comment (morgante) [inline catalog/gke/cluster/setters.yaml:25]: We should move shared setters to the top-level package. The way it's currently set up you'll have to update the value in each subpackage.

**Review by morgante — CHANGES_REQUESTED** [review]

File: catalog/gke/cluster/setters.yaml (hunk 2 above)
Comment (karlkfi) [reply catalog/gke/cluster/setters.yaml:25]: I tested it. It works as-is. 

The inner pipelines run before the outer pipelines, which means inner values only matter for setters that aren't in the outer values.

I filed a FR to provide what I think would be more intuitive behavior: https://github.com/GoogleContainerTools/kpt/issues/2385

File: catalog/gke/cluster/README.md (hunk 1 above)
Comment (karlkfi) [reply catalog/gke/cluster/README.md:17]: Yeah, the readmes maybe don't make much sense without a replacement.
I could just remove them and link to the setters.yaml tho.

File: catalog/gke/cluster/setters.yaml (hunk 2 above)
Comment (morgante) [reply catalog/gke/cluster/setters.yaml:25]: I see. That's definitely not intuitive but we don't need to block on it.

**Review by morgante — APPROVED** [review]

File: catalog/gke/cluster/README.md (hunk 1 above)
Comment (karlkfi) [reply catalog/gke/cluster/README.md]: Updated using https://github.com/GoogleCloudPlatform/blueprints/pull/42

**Pull request merged**